frontend/alert/components/AutoComplete.tsx

```diff
@@ -238,12 +159,44 @@ const AutoComplete = ({ defaultValue="", onValueChange, setTimeline, disabled })
     const [backdrop, setBackdrop] = useState("");
 
     const show = active && suggestions.length > 0;
+    const bulkMode = selectedCourses.size > 1;
 
     const setValue = (v) => {
-        onValueChange(v);
         return setInternalValue(v);
     };
 
+    useEffect(() => {
+        // Don't run if first render to prevent defaultValue of value from emptying
+        if (initialRender.current) {
+            initialRender.current = false;
+            return;
+        }
+
+        // update suggestions and input value
+        if (inputRef.current) {
+            if (bulkMode) {
+                // display all sections selected
+                inputRef.current.value = generateCoursesValue();
+            } else if (selectedCourses.size == 1) {
+                // display the one selected section
+                setInternalValue(
+                    selectedCourses.values().next().value.section_id
+                );
+                inputRef.current.value = selectedCourses
+                    .values()
+                    .next().value.section_id;
+            } else if (!inputValDelete.current) {
+                // No sections selected, clear input and suggestions
+                // Run when the only course that was selected is unselected
+                inputRef.current.value = "";
+                setInternalValue("");
+            } else {
+                inputValDelete.current = false;
+            }
+        }
+    }, [selectedCourses]);
```

Comment:
Noting that this is safe because we have the invariant that we only change `selectedCourses` by creating a new set, rather than mutating it (useEffect checks for referential changes in its dependencies).